weblate/machinery/apertium.py

```diff
@@ -91,6 +94,27 @@ def map_language_code(self, code):
             return LANGUAGE_MAP[code]
         return code
 
+    def validate_settings(self) -> None:
+        try:
+            languages = self.download_languages()
```

Comment:
This can raise any error, these should be handled similarly as in parent class (and outside the IndexError block).

Reply:
Yes. That part is on line 105 (`except Exception as error:`) same as parent class but
```py
# languages = []
source, language = languages[0]
```
will raise `IndexError`, if there are no supported languages at all.

Reply:
In parent class all  exceptions from dowload_languages are catched and reported to user, here not. It should be like:

```
try:
   languages = download_languages()
except Exception:
    raise ....
try: 
    source, language = languages[0]
except IndexError:
    raise ...
```

Reply:
Right. That is more explicit. Sorry for confusion.